routers/repo/setting_protected_branch.go

```diff
@@ -216,6 +226,7 @@ func SettingsProtectedBranchPost(ctx *context.Context, f auth.ProtectBranchForm)
 		protectBranch.WhitelistDeployKeys = f.WhitelistDeployKeys
 
 		protectBranch.RequiredApprovals = f.RequiredApprovals
+		protectBranch.EnableApprovalsWhitelist = f.EnableApprovalsWhitelist
```

Comment:
Maybe if `EnableApprovalsWhitelist` or `EnableWhitelist` change to false the existing whitelist should be blanked? It may be a feature, but it can also look like a bug. 😁

Reply:
I like the current "feature" but doesn't have a strong opinion. If we change shouldn't all dependent settings be cleared? So e.g. CanPush=false will clear/false all values regarding pushing, RequiredApprovals=0 will clear/false all values regarding approving?

Reply:
> 
> 
> I like the current "feature" but doesn't have a strong opinion. If we change shouldn't all dependent settings be cleared? So e.g. CanPush=false will clear/false all values regarding pushing, RequiredApprovals=0 will clear/false all values regarding approving?

I don't have a strong opinion either, but what you've said makes sense.

Reply:
I think it makes sense to clear the settings when disabled, so I changed it. However for RequiredApprovals=0 I didn't clear the approval settings, for two reasons: 
- You might want to block at requested changes (as discussed in other issue) even if you don't require approvals.
- Sometimes I find it convenient to temporary set the number of approvals to zero 😊 .